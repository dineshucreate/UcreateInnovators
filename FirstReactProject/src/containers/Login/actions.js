
import {
     REQUEST_SUCCESS,
     REQUEST_CALL_LOGIN_API,
  } from './constants';
  
export const loginSuccess = (simiData) => (
  {
    type: REQUEST_SUCCESS,
    simiData
  }
);

export const loginRequest = (username, password) => (
  {
    type: REQUEST_CALL_LOGIN_API,
    username, 
    password, 
  }
);

