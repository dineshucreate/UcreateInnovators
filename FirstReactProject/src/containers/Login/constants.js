export const REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS';
export const REQUEST_CALL_LOGIN_API = 'LOGIN_API_REQUEST';
