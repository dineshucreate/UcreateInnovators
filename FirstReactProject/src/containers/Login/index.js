import React, { Component } from 'react';
import { Platform, StyleSheet, Text, View, ActivityIndicator, Button, Alert, ImageBackground, Image, ScrollView, TextInput, InputAccessoryView, TouchableHighlight, TouchableOpacity, FlatList, AsyncStorage } from 'react-native';
import styles from './style';

class Login extends Component {
    constructor(){
        super()
        this.state = {
            username: 'simerjeet-ucreate',
            password: 'Ucreate@123',
        }
    }
    static navigationOptions = {
        drawerLabel: 'Login',
        drawerIcon: ({ tintColor }) => (
          <Image 
            source = {{uri: 'https://facebook.github.io/react-native/docs/assets/favicon.png'}}
            style={[styles.icon, {tintColor: tintColor}]}
          />
        ),
      };

    static navigationOptions = {
        title: 'LOGIN',
    };
    
    render() {
        const { navigation } = this.props;
        const { username,password } = this.state;
        return (
              <ScrollView contentContainerStyle={styles.contentContainer}>
               <TouchableOpacity onPress = {() => this.props.navigation.toggleDrawer()}>
                    <View>
                        <Image style = {styles.profileImage}
                             source = {{uri: 'https://facebook.github.io/react-native/docs/assets/favicon.png'}}
                        />
                    </View>
                </TouchableOpacity>
                <View style={styles.container}>
                <Text style={styles.welcome}>LOGIN SCREEN</Text>
                <Text style={styles.heading}>Login</Text>
                <TextInput style = {styles.input}
                    underlineColorAndroid="transparent"
                    placeholder="Email"
                    placeholderTextColor="#9a73ef"
                    autoCapitalize="none"
                    onChangeText = {(text) => this.setState({username:text})}
                />
                <Text style = {styles.heading}>Password</Text>
                <TextInput style={styles.input}
                    underlineColorAndroid="transparent"
                    placeholder="Password"
                    placeholderTextColor="#9a73ef"
                    autoCapitalize="none"
                    onChangeText={(text) => this.setState({password:text})}
                />
                <Button  style = {styles.buttonStyle}
                title="Login"
                onPress ={ () => {
                    navigation.navigate('Home',{user :username,pwd :password})
                }} 
                />
            </View>
            </ScrollView>
        );
    }
}
export default Login;