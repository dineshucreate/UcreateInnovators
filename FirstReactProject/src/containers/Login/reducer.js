import {
     REQUEST_SUCCESS
  } from './constants';

const INITIAL_STATE = {
  loading: true,

};
export default loginReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case REQUEST_SUCCESS: {
    return { 
      ...state,
      loading: false
    };  
  }
    default:
    return state;
  }
};
