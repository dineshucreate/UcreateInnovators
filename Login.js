import React from 'react';
import { View, Text, TextInput, TouchableOpacity } from 'react-native';


export default class Login extends React.Component {

    state = {
        email: '',
        password: '',
        isLogined: false
    }

    handleUsernameChange = (text) => {
        this.setState({
            email: text
        })
    }
    handlePasswordChange = (text) => {
        this.setState({
            password: text
        })
    }

    sumOfNumber = (a, b) => {
        return a + b;
    }

    onSubmit = () => {

        if (this.state.email === 'amanpreet@gmail.com' && this.state.password === '123456') {
            this.setState({ isLogined: true })
        }
    }

    render() {
        return (
            <View>
                <TextInput
                    testID={'login-email'}
                    placeholder={'Username'}
                    onChangeText={this.handleUsernameChange}
                />
                <TextInput
                    testID={'login-password'}
                    placeholder={'Password'}
                    onChangeText={this.handlePasswordChange}
                />
                <TouchableOpacity testID={'submit'} onPress={this.onSubmit}>
                    <Text>Sumbit</Text>
                </TouchableOpacity>

            </View >
        );
    }


}