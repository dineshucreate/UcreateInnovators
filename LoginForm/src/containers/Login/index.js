import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  Image,
  TextInput
} from 'react-native';
import StyleSheet from StyleSheet;

export default class App extends Component {
    render() {
      return (
        <View style={styles.MainContainer}>
  
          <View style={styles.ImageContainer}>
            <Image style={{ width: 250, height: "50%", resizeMode: 'contain' }}
              source={require('./src/assets/ic_home_golfclub.png')} />
          </View>
  
          <View style={styles.container}>
          
          <TextInput
            placeholder="Username/UserID"
            style={styles.TextInputStyle}
            maxLength={15} />
        </View>
  
        </View >
      );
    }
  }