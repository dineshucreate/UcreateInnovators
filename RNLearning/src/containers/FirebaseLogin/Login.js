import React from 'react';
import { StyleSheet, Text, TextInput, View, Button } from 'react-native';
import firebase from 'react-native-firebase';
import LoadingView from '../../utilities/loaderView';

export default class Login extends React.Component {

  state = { email: 'csegurpreet@gmail.com', password: '12345678', errorMessage: null }

  handleLogin = () => {
    const { email, password } = this.state;
    const { navigation } = this.props;
    this.refs.loader.showModalView();
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => { 
          navigation.navigate('Main'); 
          this.refs.loader.hideModalView();
        })
      .catch(error => { 
          this.setState({ errorMessage: error.message });
          this.refs.loader.hideModalView();
      });
  }

  renderLoadingView() {
    return (
        <LoadingView ref={'loader'} message='Logging In...' parentList={this} />
    );
  }

  render() {
      const { errorMessage, email, password } = this.state;
      const { navigation } = this.props;
    return (
      <View style={styles.container}>
        {errorMessage &&
          <Text style={{ color: 'red' }}>
            {errorMessage}
          </Text>}
        <TextInput
          style={styles.textInput}
          autoCapitalize="none"
          placeholder="Email"
          onChangeText={text => this.setState({ email: text })}
          value={email}
          clearButtonMode='always'
        />
        <TextInput
          secureTextEntry
          style={styles.textInput}
          autoCapitalize="none"
          placeholder="Password"
          onChangeText={text => this.setState({ password: text })}
          value={password}
          clearButtonMode='always'
        />
        <Button title="Login" onPress={this.handleLogin} />
        <Button
          title="Don't have an account? Sign Up"
          onPress={() => navigation.navigate('SignUp')}
        />
        {
            this.renderLoadingView()
        }
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textInput: {
    height: 40,
    width: '90%',
    borderColor: 'black',
    borderWidth: 1.5,
    borderRadius: 20,
    margin: 8,
    padding: 8
  }
});
