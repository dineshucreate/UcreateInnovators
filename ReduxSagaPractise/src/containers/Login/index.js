import React, { Component } from 'react';
import { Text, View, ImageBackground, TextInput, Image } from 'react-native';
import { loginRequest } from './action'
import PropTypes from 'prop-types';
import styles from './style';
import Buttonlogin from '../../styling/button';
import Loader from '../loader/index';
import { connect } from 'react-redux';

class Login extends Component {
    constructor(props) {
        super(props)
        this.state = {
            email: 'y@y.co',
            password: 'Ios@1234',
            loading: false,
            erremail: '',
            errpassword: ''
        }
    }

   
    onLoginPress = () => {

        const { loginRequest, navigation } = this.props;
        const { email, password, loading } = this.state
        loginRequest(email, password, navigation);
    }

    render() {
        const { navigate } = this.props.navigation;
        console.log('...Render.......  ' + this.props);

        return (

            <ImageBackground source={require('../../assets/global_bg.png')}
                style={styles.containerImage}>

                <View style={{ flex: 0.8, alignItems: 'center', justifyContent: 'center' }}>
                    <Image style={{ width: 250, height: 130, resizeMode: 'contain' }}
                        source={require('../../assets/ic_im_logo.png')} />

                    <View style={styles.welcome}>

                        <TextInput style={styles.input}
                            underlineColorAndroid="#06878A"
                            placeholder="Email"
                            placeholderTextColor="#06878A"
                            autoCapitalize="none"
                            onChangeText={email => this.setState({ email })}
                        />
                        <TextInput style={styles.input}
                            underlineColorAndroid="#06878A"
                            placeholder="Password"
                            placeholderTextColor="#06878A"
                            autoCapitalize="none"
                            onChangeText={password => this.setState({ password })}
                        />
                        <Text style={styles.titleText}>Forgot Password</Text>

                    </View>
                    <Buttonlogin title="Login"
                        onClick={this.onLoginPress}
                    />
                    <Text style={styles.TextShadowStyle}>CREATE AN ACCOUNT</Text>

                    {this.props.loading && (
                        <Loader
                            loading={this.props.loading} />
                    )}


                </View>
            </ImageBackground>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        loading: state.login.loading,
        loginData: state.login.loginData,
    };
};
const mapDispatchToProps = (dispatch) => ({
    loginRequest: (email, password, navigator) => dispatch(loginRequest(email, password, navigator))

});

export default connect(mapStateToProps, mapDispatchToProps)(Login);