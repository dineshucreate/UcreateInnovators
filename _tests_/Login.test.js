import React from 'react';
import { shallow } from 'enzyme';
import 'react-native';
import Login from '../Login';
import toJson from 'enzyme-to-json';



describe('Login Flow', () => {
    let wrapper = shallow(<Login />);

    let email = 'amanpreet@gmail.com';
    let password = '123456';

    it('create snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // it('email check', () => {
    //     wrapper.find("[testID='login-email']").simulate('ChangeText', 'aman@gmail.com');
    //     stateEmail = wrapper.state('email');
    //     var mailformat = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    //     expect(stateEmail).not.toEqual('');
    //     expect(stateEmail).toMatch(mailformat);
    //     //  expect(stateEmail).toEqual(email);
    // })

    // it('password check', () => {
    //     wrapper.find("[testID='login-password']").simulate('ChangeText', password);
    //     expect(wrapper.state('password')).not.toEqual('');
    //     expect(wrapper.state('password')).toBeString();
    //     expect(wrapper.state('password')).toEqual(password);
    // })

    // it('login check with right data', () => {
    //     wrapper.find("[testID='submit']").simulate('press');
    //     expect(wrapper.state('isLogined')).toBe(true);
    // })

    it('Sum of two numbers', () => {
        let instance = wrapper.instance();
        expect(instance.sumOfNumber(2, 3)).toBe(5);
    });
});