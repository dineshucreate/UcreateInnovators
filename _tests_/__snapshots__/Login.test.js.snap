// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Flow create snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Username"
    testID="login-email"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    testID="login-password"
    underlineColorAndroid="transparent"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    testID="submit"
  >
    <Text>
      Sumbit
    </Text>
  </TouchableOpacity>
</View>
`;
